### PR TITLE
Men 3679: Document all the tools available in the Mender-docs repository

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

File: scripts/Readme-weasel.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# weasel.sh - A simple weasel-word checker
```
Comment (mirzak) [inline scripts/Readme-weasel.md:1]: I would merge this with `Readme-passive.md `. Maybe call it,

```
README-tools.markdown
```

In lack of better name suggestion :)

File: scripts/Readme-weasel.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```bash
+$ mender-docs: bash ./scripts/weasel.sh 05.Client-configuration/03.Identity/docs.md 
+05.Client-configuration/03.Identity/docs.md:30:very
+05.Client-configuration/03.Identity/docs.md:37:several
+05.Client-configuration/03.Identity/docs.md:57:several
+05.Client-configuration/03.Identity/docs.md:74:very
+05.Client-configuration/03.Identity/docs.md:80:Clearly
+[OK] The weasel rate for 05.Client-configuration/03.Identity/docs.md:
+	weasel-words/total-lines = 5/131 ~= 3 - and thus below the cut-off rate: 3
+```
```
Comment (mirzak) [inline scripts/Readme-weasel.md:28]: Typically would use the following when code block is "output" and not to be executed:

```bash
>```bash
>code
>```
```

Then you get somehing like this:

>```bash
>mender-docs: bash ./scripts/weasel.sh 05.Client-configuration/03.Identity/docs.md 
>05.Client-configuration/03.Identity/docs.md:30:very
>05.Client-configuration/03.Identity/docs.md:37:several
>05.Client-configuration/03.Identity/docs.md:57:several
>05.Client-configuration/03.Identity/docs.md:74:very
>05.Client-configuration/03.Identity/docs.md:80:Clearly
>[OK] The weasel rate for 05.Client-configuration/03.Identity/docs.md:
>	weasel-words/total-lines = 5/131 ~= 3 - and thus below the cut-off rate: 3
>```

Pretty :)

File: scripts/Readme-passive.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+```bash
+$ mender-docs: bash ./scripts/passive.sh 01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md
+01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:37:been converted
+01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:37:been installed
+01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:123:be asked
+01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:183:be found
+[OK] The passive rate for 01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:
+	passive-words/total-lines = 3/195 ~= 1 - and thus below the cut-off rate: 3
+
+```
```
[the remark this answers is not in the document]
Comment (mirzak) [reply scripts/Readme-passive.md:27]: Typically would use the following when code block is "output" and not to be executed:

```bash
>```bash
>code
>```
```

Then you get something like this:

>```bash
>mender-docs: bash ./scripts/passive.sh 01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md
>01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:37:been converted
>01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:37:been installed
>01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:123:be asked
>01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:183:be found
>[OK] The passive rate for 01.Getting-started/01.Quickstart-with-raspberry-pi/docs.md:
>	passive-words/total-lines = 3/195 ~= 1 - and thus below the cut-off rate: 3
>```

Pretty :)

[the remark this answers is not in the document]
Comment (mirzak) [reply scripts/Readme-passive.md:27]: Also, `01.Getting-started/01.Quickstart-with-raspberry-pi` has now been removed, so maybe update example to use something that exists in the repo to make it more relevant

File: scripts/Readme-weasel.md (hunk 1 above)
Comment (oleorhagen) [reply scripts/Readme-weasel.md:1]: Sure.

Maybe we should even make a tools directory?

Comment (mirzak) [reply scripts/Readme-weasel.md:1]: I would keep it at the top for now (for visibility)

Comment (oleorhagen) [reply scripts/Readme-weasel.md:1]: After pondering it a bit, I think I would actually favour keeping them separate.
Unless I'm facing strong objections here.

Comment (oleorhagen) [reply scripts/Readme-weasel.md]: Maybe the `README-tools.md` can just be the overview, and link to the other files?

File: README-checklinks.markdown [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+`checklinks` is a `node.js` script based upon the
+[`remark-validate-links`](https://github.com/remarkjs/remark-validate-links)
+tool. It validates all internal links (i.e., [foobar](#install)).
+
+## Install
+
+The tool has a few `node.js` requirements. If you already have `node` installed on your system, run:
+
+```bash 
+$ npm install
```
Comment (mirzak) [inline README-checklinks.markdown:12]: Remove `$`

**Comment by mirzak** [comment]
Do not forget to clean up history before merging. I assumed you would do it, so did not comment :)

**Comment by oleorhagen** [comment]
> Do not forget to clean up history before merging. I assumed you would do it, so did not comment :)

What's wrong with the history here?

**Comment by mirzak** [comment]
All of these are just fixes to the original commits that are non-existing in master.

```
Add a Readme for the 'didyoumean' script …
Review fixups …
Rename from Readme-X.md -> README-X.markdown …
Remove the $ prefix …
Change bash highlighting to console for console examples 
```

Above I consider "review history", but I think that is what the PR is for on github, and there is no need to have this information in the git history.

Cleaner would be:

```
Add a Readme for the passive.sh script …
Add a Readme for the weasel.sh checker …
Add a Readme for the checklinks script …
```

Also, https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#example :)
As this is what this PR did.

**Comment by oleorhagen** [comment]
> All of these are just fixes to the original commits that are non-existing in master.
> 
> ```
> Add a Readme for the 'didyoumean' script …
> Review fixups …
> Rename from Readme-X.md -> README-X.markdown …
> Remove the $ prefix …
> Change bash highlighting to console for console examples 
> ```
> 
> Above I consider "review history", but I think that is what the PR is for on github, and there is no need to have this information in the git history.
> 
> Cleaner would be:
> 
> ```
> Add a Readme for the passive.sh script …
> Add a Readme for the weasel.sh checker …
> Add a Readme for the checklinks script …
> ```
> 
> Also, https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#example :)

Touche!

Noted.

**Pull request merged**